### PR TITLE
If variable is not used within the loop body, start the name with an underscore

### PR DESCRIPTION
**Pull request by mozga-intel**
## Description ##
Loop control variable 'x' is not used within the loop body. If this is intended, start the name with an underscore.

### Essentials ###
- [ ] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by mozga-intel** [comment]
@szha Could you help:  

Please have a look at the test: [test](https://github.com/apache/incubator-mxnet/blob/master/tests/python/unittest/test_operator.py#L4156). There is an axis param that is set up in a loop: [loop] Loop control variable 'axis' not used within the loop body but it's only used out of the loop scope: [loop](https://github.com/apache/incubator-mxnet/blob/master/tests/python/unittest/test_operator.py#L4183)
```python
    for axis in range(-data_ndim, data_ndim):
        data_shape = ()
        for _ in range(data_ndim):
            data_shape += (np.random.randint(low=1, high=5), )
        idx_shape = ()
        for _ in range(idx_ndim):
            idx_shape += (np.random.randint(low=1, high=5), )
     ....
    
    result = mx.sym.take(a=data, indices=idx, axis=axis, mode=mode)
```
The axis is always equal to `data_ndim- 1`.  Following the documentation [doc](https://mxnet.apache.org/versions/1.6/api/r/docs/api/mx.nd.take.html), the axis of an input array is to be taken. For the input tensor of rank r, it could be in the range of [-r, r-1]. Where r = data_ndim

Could you please tell whether the lines [L4191](https://github.com/apache/incubator-mxnet/blob/master/tests/python/unittest/test_operator.py#L4191) to [L4231](https://github.com/apache/incubator-mxnet/blob/master/tests/python/unittest/test_operator.py#L4231) shouldn't be placed in the loop scope to get a changeable axis?

**Comment by mozga-intel** [comment]
@mxnet-bot run ci [website]

**Comment by mozga-intel** [comment]
@mxnet-bot run ci [centos-gpu]

**Comment by mozga-intel** [comment]
@mxnet-bot run ci [website]

**Comment by mozga-intel** [comment]
@mxnet-bot run ci [website, centos-gpu]

**Comment by szha** [comment]
> Could you please tell whether the lines L4191 to L4231 shouldn't be placed in the loop scope to get a changeable axis?

I think that would make more sense than what the test currently does.

**Comment by mozga-intel** [comment]
> > Could you please tell whether the lines L4191 to L4231 shouldn't be placed in the loop scope to get a changeable axis?
> 
> I think that would make more sense than what the test currently does.

@szha Thanks! There is a pull request that includes this change: #20532

File: python/mxnet/gluon/contrib/estimator/estimator.py [hunk 1]
```diff
@@ -392,7 +392,7 @@ def fit(self, train_data,
             for handler in epoch_begin:
                 handler.epoch_begin(estimator_ref)
 
-            for i, batch in enumerate(train_data):
+            for _, batch in enumerate(train_data):
```
Comment (bgawrych) [inline python/mxnet/gluon/contrib/estimator/estimator.py:395]: in most of the cases enumerate makes no sense

**Review by bgawrych — COMMENTED** [review]
somewhere for _, _ ... can be just replaced by for _ ...

**Comment by mozga-intel** [comment]
@mxnet-bot run ci [centos-gpu, unix-gpu]

File: python/mxnet/gluon/contrib/estimator/estimator.py (hunk 1 above)
Comment (szha) [reply python/mxnet/gluon/contrib/estimator/estimator.py:395]:
```suggestion
            for batch in train_data:
```

File: tests/python/unittest/test_gluon_data.py [hunk 2]
```diff
@@ -325,14 +325,14 @@ def _batchify(data):
...
+    for _ in range(1):
+        for _ in enumerate(loader):
             pass
 
     data = _Dummy(True)
     loader = DataLoader(data, batch_size=40, batchify_fn=_batchify_list, num_workers=2)
-    for epoch in range(1):
-        for i, data in enumerate(loader):
+    for _ in range(1):
+        for _ in enumerate(loader):
```
Comment (szha) [inline tests/python/unittest/test_gluon_data.py:335]:
```suggestion
        for _ in loader:
```

File: tests/python/unittest/test_gluon_data.py [hunk 3]
```diff
@@ -382,25 +382,25 @@ def test_dataset_filter():
...
     a_filtered = a.filter(lambda x: x % 10 == 0).__mx_handle__()
     assert(len(a_filtered) == 10)
-    for idx, sample in enumerate(a_filtered):
+    for _, sample in enumerate(a_filtered):
         assert sample % 10 == 0
     a_xform_filtered = a.transform(lambda x: x + 1).filter(lambda x: x % 10 == 0)
     assert(len(a_xform_filtered) == 10)
     # the filtered data is already transformed
-    for idx, sample in enumerate(a_xform_filtered):
+    for _, sample in enumerate(a_xform_filtered):
```
Comment (szha) [inline tests/python/unittest/test_gluon_data.py:403]:
```suggestion
    for sample in a_xform_filtered:
```

File: tests/python/unittest/test_gluon_data.py [hunk 4]
```diff
@@ -417,7 +417,7 @@ def test_dataset_shard():
     assert len(shard_3) == 2
     total = 0
     for shard in [shard_0, shard_1, shard_2, shard_3]:
-        for idx, sample in enumerate(shard):
+        for _, sample in enumerate(shard):
```
Comment (szha) [inline tests/python/unittest/test_gluon_data.py:420]:
```suggestion
        for sample in shard:
```

File: tests/python/unittest/test_gluon_data.py [hunk 5]
```diff
@@ -451,7 +451,7 @@ def test_dataset_take():
     assert len(a_take_10) == count
     expected_total = sum([i for i in range(count)])
     total = 0
-    for idx, sample in enumerate(a_take_10):
+    for _, sample in enumerate(a_take_10):
```
Comment (szha) [inline tests/python/unittest/test_gluon_data.py:454]:
```suggestion
    for sample in a_take_10:
```

File: tests/python/unittest/test_gluon_data.py [hunk 6]
```diff
@@ -460,7 +460,7 @@ def test_dataset_take():
     assert len(a_xform_take_10) == count
     expected_total = sum([i * 10 for i in range(count)])
     total = 0
-    for idx, sample in enumerate(a_xform_take_10):
+    for _, sample in enumerate(a_xform_take_10):
```
Comment (szha) [inline tests/python/unittest/test_gluon_data.py:463]:
```suggestion
    for sample in a_xform_take_10:
```

File: tests/python/unittest/test_gluon_data.py [hunk 7]
```diff
@@ -477,7 +477,7 @@ def test_dataset_take_handle():
     assert len(a_take_10) == count
     expected_total = sum([i for i in range(count)])
     total = 0
-    for idx, sample in enumerate(a_take_10):
+    for _, sample in enumerate(a_take_10):
```
Comment (szha) [inline tests/python/unittest/test_gluon_data.py:480]:
```suggestion
    for sample in a_take_10:
```

File: tests/python/unittest/test_gluon_data.py [hunk 8]
```diff
@@ -486,7 +486,7 @@ def test_dataset_take_handle():
     assert len(a_xform_take_10) == count
     expected_total = sum([i for i in range(count)])
     total = 0
-    for idx, sample in enumerate(a_xform_take_10):
+    for _, sample in enumerate(a_xform_take_10):
```
Comment (szha) [inline tests/python/unittest/test_gluon_data.py:489]:
```suggestion
    for sample in a_xform_take_10:
```

File: tests/python/unittest/test_gluon_probability_v2.py [hunk 9]
```diff
@@ -1842,7 +1842,7 @@ def forward(self, loc, scale, *args):
                             rtol=1e-3, use_broadcast=False)
 
     # Test sampling
-    for shape, hybridize in itertools.product(shapes, [True, False]):
+    for shape, _ in itertools.product(shapes, [True, False]):
```
Comment (szha) [inline tests/python/unittest/test_gluon_probability_v2.py:1845]: I think hybridize should be kept and on line 1854 insert the following
```python
        if hybridize:
            net.hybridize()
```

File: tests/python/unittest/test_numpy_interoperability.py [hunk 10]
```diff
@@ -421,7 +421,7 @@ def _add_workload_swapaxes():
     # assertRaises(np.AxisError, np.swapaxes, -5, 0)
     for i in range(-4, 4):
         for j in range(-4, 4):
-            for k, src in enumerate((a, b)):
+            for _, src in enumerate((a, b)):
```
Comment (szha) [inline tests/python/unittest/test_numpy_interoperability.py:424]:
```suggestion
            for src in (a, b):
```

File: tests/python/unittest/test_numpy_op.py [hunk 11]
```diff
@@ -4846,7 +4846,7 @@ def forward(self, param1, param2):
     dtypes = ['float16', 'float32', 'float64']
     op_names = ['uniform_n', 'normal_n']
 
-    for bshape, eshape, dtype, op in itertools.product(batch_shapes, event_shapes, dtypes, op_names):
+    for bshape, eshape, _, op in itertools.product(batch_shapes, event_shapes, dtypes, op_names):
```
Comment (szha) [inline tests/python/unittest/test_numpy_op.py:4849]: dtype should be added to line 4834 and 4842 so that different data types are actually tested.

File: tests/python/unittest/test_numpy_op.py [hunk 12]
```diff
@@ -8396,7 +8396,7 @@ def dbg(name, data):
                                           dtype=dtype))
                 for optimize in [False, True]:
                     x = []
-                    for (iop, op) in enumerate(operands):
+                    for (iop, _) in enumerate(operands):
```
Comment (szha) [inline tests/python/unittest/test_numpy_op.py:8399]:
```suggestion
                    for iop in range(len(operands)):
```

File: tests/python/unittest/test_numpy_op.py [hunk 13]
```diff
@@ -8410,10 +8410,10 @@ def dbg(name, data):
                     assert_almost_equal(out_mx.asnumpy(), expected_np, rtol=rtol, atol=atol)
                     out_mx.backward()
                     cur_grad = []
-                    for (iop, op) in enumerate(x):
+                    for (_, op) in enumerate(x):
```
Comment (szha) [inline tests/python/unittest/test_numpy_op.py:8413]:
```suggestion
                    for op in x:
```

File: tests/python/unittest/test_numpy_op.py [hunk 14]
```diff
@@ -8410,10 +8410,10 @@ def dbg(name, data):
...
                         cur_grad.append(op.grad.asnumpy())
                     grad.append(cur_grad)
-                for (iop, op) in enumerate(grad[0]):
+                for (iop, _) in enumerate(grad[0]):
```
Comment (szha) [inline tests/python/unittest/test_numpy_op.py:8416]:
```suggestion
                for iop in range(len(grad[0])):
```

**Review by szha — COMMENTED** [review]
Thanks for the fix. A few loops can be updated to be more clear.
